examples/xpu/simple-fence/simple-fence.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+  Copyright (c) 2023, Intel Corporation
+
+  SPDX-License-Identifier: BSD-3-Clause
+*/
+
+#include <algorithm>
+#include <cmath>
+#include <iomanip>
+#include <iostream>
+#include <chrono>
+#include <thread>
+
+// ispcrt
+#include "ispcrt.hpp"
+
+std::ostream &operator<<(std::ostream &os, const ISPCRTDeviceType dt) {
+    switch (dt) {
+    case ISPCRT_DEVICE_TYPE_AUTO:
+        os << "Auto";
+        break;
+    case ISPCRT_DEVICE_TYPE_GPU:
+        os << "GPU";
+        break;
+    case ISPCRT_DEVICE_TYPE_CPU:
+        os << "CPU";
+        break;
+    default:
+        break;
+    }
+    return os;
+}
+
+struct Parameters {
+    float *vin;
+    float *vout;
+    int count;
+};
+
+void simple_CPU_validation(std::vector<float> vin, std::vector<float> &vgold, const unsigned int SIZE) {
+    for (unsigned int i = 0; i < SIZE; i++) {
+        float v = vin[i];
+        for (int i = 0; i < 100000; i++) {
+            if (v < 3.)
+                v = v * v;
+            else
+                v = std::sqrt(v);
+        }
+
+        vgold[i] = v;
+    }
+}
+
+#define EPSILON 0.01f
+bool validate_result(std::vector<float> vout, std::vector<float> vgold, const unsigned int SIZE) {
+    bool bValid = true;
+    for (unsigned int i = 0; i < SIZE; i++) {
+        float delta = (float)fabs(vgold[i] - vout[i]);
+        if (delta > EPSILON) {
+            std::cout << "Validation failed on i=" << i << ": vout[i] = " << vout[i] << ", but " << vgold[i]
+                      << " was expected\n";
+            bValid = false;
+        }
+    }
+    return bValid;
+}
+
+static int run(const ISPCRTDeviceType device_type, const unsigned int SIZE) {
+    std::vector<float> vin(SIZE);
+    std::vector<float> vout(SIZE);
+    std::vector<float> vgold(SIZE);
+
+    ispcrt::Device device(device_type);
+
+    // Setup input array
+    ispcrt::Array<float> vin_dev(device, vin);
+
+    // Setup output array
+    ispcrt::Array<float> vout_dev(device, vout);
+
+    // Setup parameters structure
+    Parameters p;
+
+    p.vin = vin_dev.devicePtr();
+    p.vout = vout_dev.devicePtr();
+    p.count = SIZE;
+
+    auto p_dev = ispcrt::Array<Parameters>(device, p);
+
+    // Create module and kernel to execute
+    ispcrt::Module module(device, "xe_simple-fence");
+    ispcrt::Kernel kernel(device, module, "simple_ispc");
+
+    // Create task queue and execute kernel
+    ispcrt::CommandQueue queue(device, 0);
+    ispcrt::CommandList cmds = queue.createCommandList();
+
+    std::generate(vin.begin(), vin.end(), [i = 0]() mutable { return i++; });
+
+    // Calculate gold result
+    simple_CPU_validation(vin, vgold, SIZE);
```

Comment:
I think it would be reasonable to calculate after the submission, before waiting for the fence result - to demonstrate that host can do useful job before waiting for the result. How do you like it?

Reply:
Sounds good, moved after `submit` call.